src/modules/Elsa.JavaScript/Services/JintJavaScriptEvaluator.cs

```diff
@@ -176,9 +180,21 @@ private static void CreateActivityOutputAccessors(Engine engine, ExpressionExecu
             engine.SetValue($"get{outputName}From{activityOutput.ActivityName}", (Func<object?>)(() => context.GetOutput(activityOutput.ActivityId, outputName)));
     }
 
-    private static object? ExecuteExpressionAndGetResult(Engine engine, string expression)
+    private object? ExecuteExpressionAndGetResult(Engine engine, string expression)
     {
-        var result = engine.Evaluate(expression);
+        var cacheKey = "jint:script:" + expression.GetHashCode(StringComparison.Ordinal);
+
+        var parsedScript = _memoryCache.GetOrCreate(cacheKey, entry =>
+        {
+            if (_jintOptions.ScriptCacheTimeout.HasValue)
```

Comment:
If no cache timeout is given the item will stay in the cache indefinitly?

Reply:
Yes. And this will only happen if the user explicitly provides a null value for the timeout setting. In other words, out of the box, there will always be a timeout (which defaults to 1 day).